config/crd/bases/app.k8s.io_applications.yaml

```diff
@@ -72,23 +72,26 @@ spec:
                 description: AssemblyPhase represents the current phase of the application's
                   assembly. An empty value is equivalent to "Succeeded".
                 type: string
-              componentKinds:
-                description: ComponentGroupKinds is a list of Kinds for Application's
-                  components (e.g. Deployments, Pods, Services, CRDs). It can be used
-                  in conjunction with the Application's Selector to list or watch
-                  the Applications components.
+              componentGVKs:
```

Comment:
yeah, this is definitely breaking for existing data/clients

Reply:
Currently, KALM uses controller-runtime v0.19.0, which can infer the component's version from RESTMapping, but when we update controller-runtime to the latest version, the `version` field is required for an object: https://github.com/kubernetes-sigs/controller-runtime/pull/812. If we don't provide the version, we will get an error `unstructured object has no version`.

Reply:
What about we keep the original name `componentKinds`, but add an optional `version` field? 
This is tricky because if we re-generate the manifests, the version field will become required again.

Reply:
If version inference is no longer built into controller-runtime, hoisting the version inference out into your own code would be better than baking it into the version reference here. See comment at https://github.com/kubernetes-sigs/application/pull/217/files#r655813175